### PR TITLE
Remove status field from all responses.

### PR DESCRIPTION
**Pull request by piohei**
## Motivation

Status field is no longer used and removing it will help to simplify the project.

## Solution

Removed status field from all responses.
Removed mined tree (as no longer used).
The source of leaf ids for recovery is now a database instead of processed tree.
Tests are changed to not use status field:
* for on-chain mode identity is considered processed when root is returned for `/inclusionProof` request and can be verified on chain using `queryRoot` contract method (mocked chain for testing)
* for off-chain mode identity is considered processed when root is returned for `/inclusionProof` request (no checking on chain)

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation

### CODE REVIEW

File: src/server/data.rs [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
-use crate::identity_tree::{
-    Hash, InclusionProof, ProcessedStatus, RootItem, Status, UnprocessedStatus,
-};
+use crate::identity_tree::{Hash, InclusionProof, RootItem};
 use crate::prover::{ProverConfig, ProverType};
 
 #[derive(Debug, Serialize, Deserialize)]
-#[serde(transparent)]
-pub struct InclusionProofResponse(pub InclusionProof);
+pub struct InclusionProofResponse {
```
Comment (Dzejkop) [inline src/server/data.rs:11]: These structs are exactly the same as the ones in `crate::identity_tree`, why not just keep them as wrappers with `#[serde(transparent)]`?

File: src/identity_tree/mod.rs [hunk 2]
```diff
@@ -561,14 +560,9 @@ impl TreeState {
 
     #[must_use]
     pub fn get_proof_for(&self, item: &TreeItem) -> (Field, InclusionProof) {
-        let (leaf, root, proof) = match item.status {
-            ProcessedStatus::Pending => self.latest.get_leaf_and_proof(item.leaf_index),
-            ProcessedStatus::Processed => self.processed.get_leaf_and_proof(item.leaf_index),
-            ProcessedStatus::Mined => self.mined.get_leaf_and_proof(item.leaf_index),
-        };
+        let (leaf, root, proof) = self.latest.get_leaf_and_proof(item.leaf_index);
```
Comment (Dzejkop) [inline src/identity_tree/mod.rs:563]: Eventually we'll also want to remove the different tree versions. But let's do it in another PR

File: tests/immediate_deletion.rs [hunk 3]
```diff
@@ -81,6 +81,7 @@ async fn immediate_deletion() -> anyhow::Result<()> {
     // Check that we can also get these inclusion proofs back.
     for i in 0..insertion_batch_size {
         test_inclusion_proof(
```
Comment (Dzejkop) [inline tests/immediate_deletion.rs:83]: This function call is missing an arg

Comment (Dzejkop) [reply tests/immediate_deletion.rs:83]: Nevermind, I think I had an outdated branch checked out

File: src/server/data.rs (hunk 1 above)
Comment (piohei) [reply src/server/data.rs:11]: I personally don't like mixing the application layers and exposing other modules as part of the module's API. This way there is no need to change code for users of the server module when InclusionProof in identity_tree will change. If you don't find it useful and wan't to keep repo smaller let me know. I will flatten it then. :)

File: src/identity_tree/mod.rs (hunk 2 above)
Comment (piohei) [reply src/identity_tree/mod.rs:524]: For first PR I removed processed and mined tree (but keeping the status field in database).

File: e2e_tests/scenarios/tests/common/chain.rs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+pub struct Chain {
+    pub private_key:      SigningKey,
+    pub identity_manager: SpecialisedContract,
+}
+
+#[instrument(skip_all)]
+pub async fn create_chain(chain_addr: String) -> anyhow::Result<Chain> {
+    // This private key is taken from tx-sitter configuration in compose.yaml.
+    // Env name: TX_SITTER__SERVICE__PREDEFINED__RELAYER__KEY_ID
+    let private_key = SigningKey::from_slice(&[
```
Comment (Dzejkop) [inline e2e_tests/scenarios/tests/common/chain.rs:35]: Can we use `hex_literal` crate?

File: e2e_tests/scenarios/tests/common/mod.rs [hunk 5]
```diff
@@ -132,28 +136,39 @@ pub async fn insert_identity_with_retries(
...
     for _i in 0..retries_count {
         last_res = inclusion_proof(client, uri, commitment).await;
 
         if let Ok(ref inclusion_proof_json) = last_res {
-            if inclusion_proof_json.0.status == Status::Processed(Mined) {
-                break;
+            if let Some(root) = inclusion_proof_json.root {
+                let res = chain
+                    .identity_manager
+                    .method::<U256, RootInfo>("queryRoot", root.into())
```
Comment (Dzejkop) [inline e2e_tests/scenarios/tests/common/mod.rs:152]: We should put these function in the abi derive - then we'll be able to just call them like methods, e.g. [like here](https://github.com/worldcoin/world-tree/blob/main/tests/full_flow.rs#L96) (except for calls you wouldn't use the `.send()`)

File: e2e_tests/scenarios/tests/common/mod.rs [hunk 6]
```diff
@@ -132,28 +136,39 @@ pub async fn insert_identity_with_retries(
...
 
         _ = sleep(Duration::from_secs_f32(retries_interval)).await;
     }
 
-    let inclusion_proof_json = last_res?;
-
-    assert_eq!(inclusion_proof_json.0.status, Status::Processed(Mined));
+    if last_res.is_err() {
+        return last_res.map(|_| ());
+    }
```
Comment (Dzejkop) [inline e2e_tests/scenarios/tests/common/mod.rs:169]: Isn't this just
```suggestion
    last_res?;
```

File: src/server/data.rs (hunk 1 above)
Comment (Dzejkop) [reply src/server/data.rs:11]: Let's keep things simple, let's reuse the structs from `crate::identity_tree`

Comment (Dzejkop) [reply src/server/data.rs:11]: Same for other structs

Comment (piohei) [reply src/server/data.rs]: done

File: e2e_tests/scenarios/tests/common/chain.rs (hunk 4 above)
Comment (piohei) [reply e2e_tests/scenarios/tests/common/chain.rs]: done

File: e2e_tests/scenarios/tests/common/mod.rs (hunk 5 above)
Comment (piohei) [reply e2e_tests/scenarios/tests/common/mod.rs]: done

File: tests/common/chain_mock.rs [hunk 7]
```diff
@@ -206,14 +207,16 @@ pub async fn spawn_mock_chain(
         .await?
         .await?;
```
Comment (piohei) [inline tests/common/chain_mock.rs:208]: For some reason changing this call to:
```
identity_manager
     .initialize_v2(delete_verifiers.address())
     .call()
     .await?;
 ```
doesn't work. It compiles but then transactions are being failed. I don't want now to focues on that so just changing other calls.

**Review by Dzejkop — APPROVED** [review]
LGTM
